framework/src/org/checkerframework/framework/test/ImmutableTestConfiguration.java

```diff
@@ -0,0 +1,87 @@
+package org.checkerframework.framework.test;
+
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents all of the information needed to execute the Javac compiler for a given set of test files.
+ */
+public class ImmutableTestConfiguration implements TestConfiguration {
+
+    /**
+     * Options that should be passed to the compiler
```

Comment:
SimpleOptionMap is a utility class that contains a Map like above and a set of utility methods that manipulate the map.  SimpleOptionMap contains methods similar to addToClasspath above except they work like:  addToPath("-cp", "myClassesDir").  Then the ConfigurationBuilder's method looks as follows:
public void addToClasspath(String classesDir) {
   optionMap.addToPath("-cp", "myClassesDir");
}